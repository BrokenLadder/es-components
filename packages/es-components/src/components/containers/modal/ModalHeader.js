import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { noop } from 'lodash';
import DismissButton from '../../controls/DismissButton';

// Note: ModalHeader relies on a parent (Modal) with ThemeProvider wrapping it
const Header = styled.div`
  border-bottom: 2px solid ${props => props.theme.brandColors.vbMagenta};
  color: ${props => props.theme.colors.gray9};
  display: flex;
  font-size: 26px;
  font-weight: 500;
  justify-content: space-between;
  padding: 15px;
`;

const Title = styled.h4`
  font-size: 24px;
  font-weight: 500;
  margin: 0;
`;

const DismissModal = styled(DismissButton)`
  color: ${props => props.theme.colors.black};
  line-height: 1;
  opacity: 0.2;
  text-shadow: 0 1px 0 ${props => props.theme.colors.white};

  &:hover {
    opacity: 0.5;
  }
`;

const ModalHeader = (props, { modal }) => {
  const { hideCloseButton, children, ...otherProps } = props;
  const onHide = modal ? modal.onHide : noop;
  const ariaId = modal ? modal.ariaId : null;

  return (
    <Header {...otherProps}>
      <Title id={ariaId}>{children}</Title>

      {!hideCloseButton && <DismissModal onClick={onHide} />}
    </Header>
  );
};

ModalHeader.propTypes = {
  /** Specify whether the modal header should contain a close button */
  hideCloseButton: PropTypes.bool,
  children: PropTypes.any
};

ModalHeader.defaultProps = {
  hideCloseButton: false,
  children: null
};

ModalHeader.contextTypes = {
  modal: PropTypes.shape({
    onHide: PropTypes.func,
    ariaId: PropTypes.string
  })
};

export default ModalHeader;
