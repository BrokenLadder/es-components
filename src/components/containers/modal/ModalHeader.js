import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colors } from '../../theme';
import { noop } from 'lodash';

const Header = styled.div`
  background-color: ${colors.accent};
  border-bottom: 1px solid ${colors.grayLighter};
  color: ${colors.white};
  display: flex;
  justify-content: space-between;
  padding: 15px;
`;

const Title = styled.h4`
  font-size: 24px;
  font-weight: 500;
  margin: 0;
`;

const Button = styled.button`
  background: transparent;
  border: 0;
  color: ${colors.white};
  cursor: pointer;
  font-size: 27px;
  font-weight: bold;
  line-height: 1;
  margin-top: -2px;
  opacity: 1;
  padding: 0;
  text-shadow: 0 1px 0 ${colors.black};
`;

const ScreenReaderText = styled.span`
  border: 0;
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
`;

const ModalHeader = (props, context) => {
  const { closeButton, children, ...otherProps } = props;
  const modal = context.modal;
  const onHide = modal ? modal.onHide : noop;
  const ariaId = modal ? modal.ariaId : null;

  return (
    <Header {...otherProps}>
      <Title id={ariaId}>{children}</Title>

      {closeButton &&
        <Button type="button" aria-label="Close" onClick={onHide}>
          <span aria-hidden="true">Ã—</span>
          <ScreenReaderText>Close</ScreenReaderText>
        </Button>}
    </Header>
  );
};

ModalHeader.propTypes = {
  /** Specify whether the Component should contain a close button */
  closeButton: PropTypes.bool,
  children: PropTypes.any
};

ModalHeader.defaultProps = {
  closeButton: true
};

ModalHeader.contextTypes = {
  modal: PropTypes.shape({
    onHide: PropTypes.func,
    ariaId: PropTypes.string
  })
};

export default ModalHeader;
