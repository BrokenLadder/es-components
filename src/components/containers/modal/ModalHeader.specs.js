/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';

import ModalHeader from './ModalHeader';

describe('modal header component', () => {
  let instanceToRender;

  it('renders with close button', () => {
    instanceToRender = (
      <ModalHeader className="header">
        This is the header with close button.
      </ModalHeader>
    );

    const tree = renderer.create(instanceToRender).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without close button', () => {
    instanceToRender = (
      <ModalHeader className="header" closeButton={false}>
        This is the header without close button.
      </ModalHeader>
    );

    const tree = renderer.create(instanceToRender).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
