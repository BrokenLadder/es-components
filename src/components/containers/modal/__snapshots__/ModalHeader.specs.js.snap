exports[`modal header component renders with close button 1`] = `
<div
  className="header sc-bdVaJa UgNLO">
  <h4
    className="sc-bwzfXH dtTgkt"
    id={null}>
    This is the header with close button.
  </h4>
  <button
    aria-label="Close"
    className="sc-htpNat hloEoM"
    onClick={[Function]}
    type="button">
    <span
      aria-hidden="true">
      Ã—
    </span>
    <span
      className="sc-bxivhb COfqB">
      Close
    </span>
  </button>
</div>
`;

exports[`modal header component renders without close button 1`] = `
<div
  className="header sc-bdVaJa UgNLO">
  <h4
    className="sc-bwzfXH dtTgkt"
    id={null}>
    This is the header without close button.
  </h4>
</div>
`;
